metaflow/plugins/argo/argo_inner_flow.py

```diff
@@ -34,7 +67,104 @@ def __init__(
         self._argo_run_id = self._flow_information["metadata"]["name"]
         self._metaflow_run_id = f"argo-{self._argo_run_id}"
         self._kubernetes_namespace = self._flow_information["metadata"]["namespace"]
-        self._exception = None
-        self._status = None
 
-        print(f"Inner flow has started w/ Argo id: {self._argo_run_id}")
+        # Waiting for Argo workflow to trigger is not optional.
+        # It is necessary to determine Flow name if not given as parameter
+        wait_to_trigger = 20  # wait time is 20 seconds
+        print("attempting to trigger Argo workflow")
+        while wait_to_trigger > 0 and self.status is None:
+            time.sleep(1)
+            wait_to_trigger -= 1
+
+        if wait_to_trigger == 0:
+            raise Exception("Inner flow failed to begin running")
+
+        print(
+            f"A run of Metaflow flow {self._flow_name} has started w/ Argo id: {self._argo_run_id}"
+        )
+
+        # [optional] wait for argo workflow to finish running
+        if self._wait:
+            print("\nNow we will wait for the flow to finish")
+            wait_remaining = self._wait_to_run  # in minutes (5s for testing)
+            print(f"status: {self.status}, 'mins' to wait: {wait_remaining}")
+            while wait_remaining > 0 and self.status == "Running":
+                wait_remaining -= 1
+                print(f"status: {self.status}, 'mins' to wait: {wait_remaining}")
+                time.sleep(5)
```

Comment:
Looks like you are sleeping for 5 seconds but subtracting 1 minute?

Reply:
It was faster for debugging to use 5 seconds. I was planning to change it at the end (see my note to self on line 27). But it makes sense that I should change it now, since this is a pull request.

Reply:
done